index.src.html

```diff
@@ -1301,86 +1301,83 @@ <h4 id="monkey-patching">Monkey Patches to Fetch</h4>
   pull request against Fetch.
 
   <h4 id="body-extraction" algorithm>
-    Extract a |body| and |Content-Type| for |request|'s attached credential
+    Extract an |action| and |Content-Type| for |request|'s attached credential
   </h4>
 
-  Given a {{Request}} (|request|), this algorithm returns a {{Body}} (|body|) and
-  a string (|Content-Type|):
+  The following algorithm specializes the body mixin method to [=extract=] a
+  request body from an object, for objects of type {{PasswordCredential}}.
+  Given a {{Request}} |request| and its <a>attached credential</a>
+  |credential|, this algorithm returns an |action| and a string
+  |Content-Type|:
 
-  1.  Let |credential| be |request|'s <a>attached credential</a>.
+  1.  Let |action| be `null`, and |Content-Type| be the empty string.
 
-  2.  Let |body| be a new <a for="request">body</a>, and |Content-Type| be the empty string.
+  2.  If |request|'s <a for="request">url</a>'s scheme is not the same as
+      |request|'s <a for="request">client</a>'s {{URL/origin}}'s scheme,
+      or |request|'s <a for="request">url</a>'s host's <a>registerable
+      domain</a> is not the same as |request|'s
+      <a for="request">client</a>'s {{URL/origin}}'s host's <a>registerable
+      domain</a>, then return |action| and |Content-Type|.
 
-  3.  If |credential| is a {{PasswordCredential}}, run these substeps.
+  3.  Let |data| be a copy of |credential|'s
+      {{PasswordCredential/additionalData}} attribute if it is not
+      `undefined`, and a new {{FormData}} object otherwise.
 
-      1.  If |request|'s <a for="request">url</a>'s scheme is not the same as
-          |request|'s <a for="request">client</a>'s {{URL/origin}}'s scheme,
-          or |request|'s <a for="request">url</a>'s host's <a>registerable
-          domain</a> is not the same as |request|'s
-          <a for="request">client</a>'s {{URL/origin}}'s host's <a>registerable
-          domain</a>, then skip the remaining substeps.
+  4.  Let |list| be an empty list.
 
-      2.  Let |data| be a copy of |credential|'s
-          {{PasswordCredential/additionalData}} attribute if it is not
-          `undefined`, and a new {{FormData}} object otherwise.
+  5.  If |data| is a {{FormData}} object, then:
 
-      3.  Let |list| be an empty list.
+      1.  Set |list| to a copy of |data|'s <a for="FormData">entries</a>.
```

Comment:
I guess we don't actually need this to be a copy, since we've already copied `|data|` entirely.